writings/2021-12-10-speeding-up-evm-part-1.mdx

```diff
@@ -0,0 +1,107 @@
+
+Of course, besides those four situations, there are other countless possibilities. For example: ``{(r, w), (r, w)}``. In this situation, it is possible that we have $txn_i$ first reads $s'$ and then changes it, but the value assigned to $s'$ always equals that of $txn_j$'s assignment because of how the smart contract was written. So this effectively reduces to the ``{(r), (r)}`` case. Or this could easily go the other way and reduce to ``{(w), (r)}``,  ``{(w), (w)}``, or ``{(r), (w)}``. And even then it could be that the writer somehow doesn't change the storage's value or that the reader doesn't affect state change in the EVM (e.g., reading some value to be 2 while it was originally 1, but the only use for that storage reading was only a `require` checking if the variable is positive). The point of those examples is just to say that there are lots of specific classes of situations where our parallelization algorithm doesn't work optimally. And due to [impossibility results](https://en.wikipedia.org/wiki/Rice%27s_theorem) they have to. So this means depending on the exact structure of $\kappa$, there are lots of long-tail optimizations for us to devise vastly different parallelization algorithms for optimal performance. We will come back to exact optimizations to this in the next post.
+
+## Conclusion
+
+EVM parallelization facilitates Ethereum scalability without compromising decentralization or requiring major changes to protocol/client. It also helps to minimize the negative impact of MEV by allowing better bundle merging and production. We approach the parallelization problem by dissecting it into two parts, generating shared data access information and devising a parallelization algorithm that utilizes the information. We've seen the landscape of Ethereum scalability solutions and discussed why current parallelization tricks don't work smoothly. 
+
+## References
+1. What Makes EVM slow, https://docs.google.com/presentation/d/11DqpZP1NA24OvJe3l9yaAPavQk7luGIE5oxIe07IayA/edit#slide=id.ge4c54e2503_0_20
```

Comment:
fyi you can write references in mdx format using

`[^1]` right after something you want to footnote

`[^1]:` at the end of the doc

see Alejo's 'Quantifying Realized Extractable Value' for an example of it :D